client/order/epochqueue.go

```diff
@@ -0,0 +1,126 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package order
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"decred.org/dcrdex/dex/msgjson"
+	"decred.org/dcrdex/dex/order"
+	"github.com/decred/dcrd/crypto/blake256"
+)
+
+// EpochQueue represents a client epoch queue.
+type EpochQueue struct {
+	orders map[string]*msgjson.EpochOrderNote
+	index  []string
+	mtx    sync.Mutex
+}
+
+// NewEpochQueue creates a client epoch queue.
+func NewEpochQueue() *EpochQueue {
+	return &EpochQueue{
+		orders: make(map[string]*msgjson.EpochOrderNote),
+		index:  make([]string, 0),
+	}
+}
+
+// Reset clears the epoch queue. This should be called when a new epoch begins.
+func (eq *EpochQueue) Reset() {
+	eq.mtx.Lock()
+	eq.orders = make(map[string]*msgjson.EpochOrderNote)
+	eq.index = make([]string, 0)
+	eq.mtx.Unlock()
+}
+
+// Queue appends the provided note to the epoch queue.
+func (eq *EpochQueue) Queue(note *msgjson.EpochOrderNote) {
+	eq.mtx.Lock()
+	defer eq.mtx.Unlock()
+	id := note.OrderID.String()
+	eq.orders[id] = note
+
+	// Update the queue sort order per order ids.
+	i := sort.Search(len(eq.index), func(i int) bool { return strings.Compare(eq.index[i], id) > 0 })
```

Comment:
Looks like your sorting on order ID, but the spec says the checksum is generated from sorted commitment hashes.